drydock/templates/drydock/k8s/drydock-jobs/mongodb.yml

```
@@ -23,12 +23,16 @@ spec:
         args:
         - |
           echo "Initialising MongoDB..."
+          {%- if MONGODB_USERNAME and MONGODB_ROOT_PASSWORD %}
           mongo --host {{MONGODB_HOST }} -u "{{ MONGODB_ROOT_USERNAME }}" -p "{{ MONGODB_ROOT_PASSWORD }}" admin <<EOF
+          {%- else %}
+          mongo --host {{MONGODB_HOST }} admin <<EOF
+          {%- endif %}
```

```suggestion 
          mongo --host {{MONGODB_HOST }} {% if MONGODB_ROOT_USERNAME and MONGODB_ROOT_PASSWORD %} -u "{{ MONGODB_ROOT_USERNAME }}" -p "{{ MONGODB_ROOT_PASSWORD }}"  admin {% endif %} <<EOF
          
```

Does this work? putting everything in a single line.

Also remember to check for `MONGODB_ROOT_USERNAME`.

---

I'm mostly meant to avoid the else block:

```
mongo --host {{MONGODB_HOST }} {% if MONGODB_ROOT_USERNAME and MONGODB_ROOT_PASSWORD %} -u {{ MONGODB_ROOT_USERNAME }} -p {{ MONGODB_ROOT_PASSWORD }} {% endif %} admin
```

If you don't set `MONGODB_ROOT_USERNAME` and `MONGODB_ROOT_PASSWORD` then the command skips this part:

```
-u {{ MONGODB_ROOT_USERNAME }} -p {{ MONGODB_ROOT_PASSWORD }}
```